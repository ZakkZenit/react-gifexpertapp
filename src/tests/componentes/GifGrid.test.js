import React from 'react';
import { shallow } from 'enzyme'
import { GifGrid } from '../../componentes/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Preubas <GifGrid />', () => {

    const category = 'Dragon Ball';

    // let wrapper = shallow( <GifGrid category={category}/> );
    
    // beforeEach( () =>  {

    //     wrapper = shallow( <GifGrid category={category}/> );

    // });
    
    test('debe cargar correctamente', () => {

        useFetchGifs.mockReturnValue({
           data: [],
           loading: true 
        });

        const wrapper = shallow( <GifGrid category={category}/> );

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe mostrar items cuando carga imagenes con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
 
        expect(wrapper).toMatchSnapshot();
    })
    

})
