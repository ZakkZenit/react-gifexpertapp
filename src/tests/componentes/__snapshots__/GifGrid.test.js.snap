// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preubas <GifGrid /> debe cargar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;

exports[`Preubas <GifGrid /> debe mostrar items cuando carga imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="cardGrid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
